checker/src/diagnostics.rs

```diff
@@ -440,12 +441,17 @@ mod defined_errors_and_warnings {
 				},
 				TypeCheckError::ReturnedTypeDoesNotMatch {
 					position,
+					returned_position,
 					expected_return_type,
 					returned_type,
-				} => Diagnostic::Position {
+				} => Diagnostic::PositionWithAdditionLabels {
 					reason: format!(
 						"Function is expected to return {expected_return_type} but returned {returned_type}",
 					),
+					labels: vec![(
```

Comment:
Thinking about this, I think `returned_position` should be the main diagnostic position. Could we swap the values for the main `position` and label?

Then the main reason should be "Cannot return {returned type} from function expected to return {expected_return_type}" and the label one should be "function annotated to return {expected_return_type} here"

Reply:
On it, let me swap them and see how the new message looks.

Reply:
How does this look?

![Screenshot 2023-11-04 150607](https://github.com/kaleidawave/ezno/assets/112388462/128d7686-7326-4b74-b315-059862338800)